packages/server/src/user/user.service.ts

```diff
@@ -1,148 +1,201 @@
-   * @returns User object
+   * @param id uuid of the user as string
+   * @returns User object, or throw `NotFoundError` when not exist
    */
-  async loginUserByUsername(params: { project_id; username; password }): Promise<USER> {
-    // TODO: Replace find temperary user
-    for (const user of this.users) {
-      if (params.project_id === user.project_id && params.username === user.username && params.password === user.password) {
-        // return dummy user for testing
-        return user;
-      }
-    }
-
-    return null;
+  async findUserById(id: string): Promise<User> {
+    return await this.prisma.user.findFirstOrThrow({ where: { id: id } });
   }
 
   /**
-   * Find unique user for a project using email and login.
+   * Find unique user for a project using email.
    *
-   * @param params object includes: `project_id`, `email`, and `password` in plain text
-   * @returns User object
+   * @returns User object, or throw `NotFoundError` when not exist
    */
-  async loginUserByEmail(params: { project_id; email; password }): Promise<USER> {
-    // TODO: Replace find temperary user
-    for (const user of this.users) {
-      if (params.project_id === user.project_id && params.email === user.email && params.password === user.password) {
-        // return dummy user for testing
-        return user;
+  async findUserByUsername(project_id: string, username: string): Promise<User> {
+    return await this.prisma.user.findFirstOrThrow({
+      where: {
+        project_id: project_id,
+        username: username
       }
-    }
+    });
+  }
 
-    return null;
+  /**
+   * Find unique user for a project using email.
+   *
+   * @returns User object, or throw `NotFoundError` when not exist
+   */
+  async findUserByEmail(project_id: string, email: string): Promise<User> {
+    return await this.prisma.user.findFirstOrThrow({
+      where: {
+        project_id: project_id,
+        email: email
+      }
+    });
   }
 
   /**
-   * Generate reset token for a user and update it in the database.
+   * Generate reset token for a user and store it in the database.
    *
-   * @param params object includes: `project_id` and `email`
-   * @param reset_code a string of reset code in plain text
+   * @param params object includes: `project_id` and `email` as string
+   * @param reset_code_plain a string of reset code in plain text
    */
-  async setResetToken(params: { project_id; email }, reset_code: string) {
-    // TODO: Replace set reset token for temperary user
-    for (let idx = 0; idx < this.users.length; idx++) {
-      const user = this.users[idx];
-      if (params.project_id === user.project_id && params.email === user.email) {
-        user.reset_code = reset_code;
-        break;
+  async setResetToken(params: { project_id; email }, reset_code_plain: string): Promise<void> {
+    const user_to_update = await this.prisma.user.findFirstOrThrow({
```

Comment:
NIT: now that you have findByEmail you can use that here instead of having the DB code.